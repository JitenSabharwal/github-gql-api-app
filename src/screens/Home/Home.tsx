import React from "react"
import Layout from "../../components/organisms/Layout/Layout"
import SearchBar from "../../components/atoms/SearchBar/SearchBar"
import { TotalCount } from "./components/TotalCount"
import {
  IssueCard,
  Issue as IssueCardType,
} from "../../components/molecules/IssueCard"

interface Issue {}
type Props = {
  totalCount: number
  issues: Issue[]
}
const cardMockData: IssueCardType = {
  id: "1",
  createdAt: new Date(),
  title: "Sample Issue",
  description:
    "This is just for testing purpose, This is just for testing purpose,This is just for testing purpose,This is just for testing purpose, This is just for testing purpose",
  createdBy: "Jiten",
  repository: "Sample Test",
  commentCount: 15,
}
export default function Home({ totalCount = 0 }: Props) {
  return (
    <Layout>
      <SearchBar />
      <TotalCount value={totalCount} label={"Total number of found issues"} />
      <IssueCard issue={cardMockData} onShowMore={console.log} />
    </Layout>
  )
}
