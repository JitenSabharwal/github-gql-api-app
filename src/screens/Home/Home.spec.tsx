import Home from "./Home"
import React from "react"
import { shallow } from "enzyme"
describe("Home", () => {
  let component: React.ReactElement
  beforeAll(() => (component = <Home totalCount={100} issues={[]} />))
  it("Should render Home Component", () => {
    const container = shallow(component)
  })
  it("Should render the layout", () => {
    const container = shallow(component)
    expect(container.getElements()).toMatchSnapshot()
  })
})
