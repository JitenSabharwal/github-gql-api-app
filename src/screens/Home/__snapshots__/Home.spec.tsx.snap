// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Should render the layout 1`] = `
Array [
  <Context.Consumer>
    [Function]
  </Context.Consumer>,
]
`;
